src/main/scala/com/databricks/labs/overwatch/utils/Config.scala

```diff
@@ -381,6 +382,43 @@ class Config() {
     }
   }
 
+  private def validateApiEnvConfig(apiEnvConfig: Option[ApiEnvConfig]) = {
+    if (apiEnvConfig.isDefined) {
+      val envConfig = apiEnvConfig.get
+      if (envConfig.threadPoolSize < 1 || envConfig.threadPoolSize > 20) {
+        throw new BadConfigException("ThreadPoolSize should be a valid number between 0 to 20")
+      }
+      if (envConfig.apiWaitingTime < 60000) { // 60000ms = 1 mint
+        throw new BadConfigException("ApiWaiting time should be more than 60000ms")
+      }
+      if (envConfig.errorBatchSize < 1 || envConfig.errorBatchSize > 1000) {
+        throw new BadConfigException("ErrorBatchSize should be between 1 to 1000")
+      }
+      if (envConfig.successBatchSize < 1 || envConfig.successBatchSize > 1000) {
+        throw new BadConfigException("SuccessBatchSize should be between 1 to 1000")
+      }
+      if (envConfig.apiProxyConfig.isDefined) {
+        validateApiEnvProxy(envConfig.apiProxyConfig)
+      }
+    }
+  }
+
+  def validateApiEnvProxy(apiProxyConfig: Option[ApiProxyConfig]) = {
```

Comment:
why did you move this to config? It doesn't belong in Config...right? I like how you switched it to validate the actual objects but no idea why you moved it here. :P let's keep this in Initializer and we'll get it cleaned up one of these days.

Reply:
The acutal object is present in Config.scala,but now Config.scala dont have access to Initialiser.scala object, we can not make call from Config.scala to Initialiser.scala methods. Thats why i moved the validation code to Config.scala.

Reply:
@Sriram-databricks but you have a getter and a setter right? If you don’t you should and then you can use that to get and validate from Initializer

Reply:
done

Reply:
NOT FOR initializer! :P -- for `ApiEnvConfig` put a getter in Config and get it into the initializer after it's set and validate it in the initializer.